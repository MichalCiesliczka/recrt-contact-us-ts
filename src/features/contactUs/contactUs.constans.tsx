export const tableColumns = [
  {
    label: 'Name',
    value: 'name'
  },
  {
    label: 'Email',
    value: 'email'
  },
  {
    label: 'Message',
    value: 'message'
  },
];

export default {
  tableColumns,
};
