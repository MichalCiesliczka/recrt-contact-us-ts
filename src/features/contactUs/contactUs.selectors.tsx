import { AppState } from "../../rootReducer";

export const getSubmittedMessages = ({ contactUs }: AppState) => contactUs.submittedMessages;

export default {
  getSubmittedMessages,
};
