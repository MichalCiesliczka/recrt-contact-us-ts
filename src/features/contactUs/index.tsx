import ContactUsContainer from './containers/contactUs';
import ContactUsComponent from './components/contactUs';

import ContactUsActions from './contactUs.actions';
import ContactUsReducer from './contactUs.reducer';

export {
  ContactUsComponent,
  ContactUsActions,
  ContactUsReducer,
};

export default ContactUsContainer;
