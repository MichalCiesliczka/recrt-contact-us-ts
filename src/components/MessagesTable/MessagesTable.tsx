import * as React from 'react';

import Table from '@material-ui/core/Table';
import TableHead from '@material-ui/core/TableHead';
import TableBody from '@material-ui/core/TableBody';

import MessagesTableHeader from './elements/MessagesTableHeader';
import MessagesTableRowContainer from './elements/MessagesTableRow';

import { SubmittedMessage, TableColumn } from '../../features/contactUs/contactUs.types';

interface Props {
  messages: SubmittedMessage[],
  columns: TableColumn[]
};

const MessagesTable: React.SFC<Props> = ({ messages, columns }) => (
  <Table>
    <TableHead>
      <MessagesTableHeader columns={columns} />
    </TableHead>
    <TableBody>
      {messages
        .map((message, index) => (
          <MessagesTableRowContainer
            key={message.id}
            message={message}
            index={index}
            columns={columns}
          />
        ))
      }
    </TableBody>
  </Table>
);

export default MessagesTable;
