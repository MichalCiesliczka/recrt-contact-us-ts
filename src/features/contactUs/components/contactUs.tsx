import * as React from 'react';

import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import ContactUsForm from '../../../components/ContactUsForm';
import MessagesTable from '../../../components/MessagesTable/MessagesTable';
import { ContactUsMessage, SubmittedMessage } from '../contactUs.types';
import { tableColumns } from '../contactUs.constans';

const styles = {
  paper: {
    margin: '30px auto',
    maxWidth: 900,
    padding: 30,
  },
  title: {
    fontSize: 30,
  },
};

interface Props {
  submitContactForm: (values: ContactUsMessage) => void,
  submittedMessages: SubmittedMessage[],
  classes: {
    paper: string,
    title: string,
  }
}

const ContactUsComponent: React.SFC<Props> = ({ submitContactForm, classes, submittedMessages }) => (
  <Paper className={classes.paper}>
    <Typography component="h1" className={classes.title}>
      Contact us!
      </Typography>
    <ContactUsForm
      saveFunction={submitContactForm}
    />
    <MessagesTable
      messages={submittedMessages}
      columns={tableColumns}
    />
  </Paper>
);

export default withStyles(styles)(ContactUsComponent);
