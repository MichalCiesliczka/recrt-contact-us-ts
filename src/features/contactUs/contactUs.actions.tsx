import { createStandardAction } from 'typesafe-actions';
import { ContactUsMessage } from './contactUs.types'

export const SUBMIT_CONTACT_FORM = 'CONTACT_FORM/SUBMIT';
export const UPDATE_MESSAGES_LIST = 'CONTACT_PAGE/UPDATE_LIST';

export const submitContactForm = createStandardAction(SUBMIT_CONTACT_FORM)<ContactUsMessage>();
export const updateMessagesList = createStandardAction(UPDATE_MESSAGES_LIST)<ContactUsMessage[]>();

export default {
  submitContactForm,
  updateMessagesList,
};
