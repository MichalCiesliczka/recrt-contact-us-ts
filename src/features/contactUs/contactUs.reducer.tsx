import { getType } from 'typesafe-actions';

import { updateMessagesList } from './contactUs.actions';
import { ContactUsState, ContactUsActions } from './contactUs.types';
import { LOCAL_STORAGE_KEY } from '../../localMessagesMiddleware';

// ------------------------------------
// State
// ------------------------------------
export const initialState: ContactUsState = {
  submittedMessages: JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY) || '""') || [],
};

// ------------------------------------
// Reducer
// ------------------------------------
export default (state: ContactUsState = initialState, action: ContactUsActions) => {
  switch (action.type) {
    case getType(updateMessagesList):
      return {
        ...state,
        submittedMessages: action.payload,
      };

    default:
      return state;
  }
}
