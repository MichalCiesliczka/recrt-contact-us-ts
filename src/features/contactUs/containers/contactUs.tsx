import { connect } from 'react-redux';

import ContactUsComponent from '../components/contactUs';
import { submitContactForm } from '../contactUs.actions';

import { AppState } from '../../../rootReducer';
import { getSubmittedMessages } from '../contactUs.selectors';

const mapStateToProps = (state: AppState) => ({
  submittedMessages: getSubmittedMessages(state),
})

const mapDispatchToProps = {
  submitContactForm,
};

export default connect(mapStateToProps, mapDispatchToProps)(ContactUsComponent);
