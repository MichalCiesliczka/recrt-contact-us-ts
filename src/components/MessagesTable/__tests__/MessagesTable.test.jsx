import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import MessagesTable from '../MessagesTable';
import { tableColumns } from '../../../features/contactUs/contactUs.constans';

describe('<MessagesTable />', () => {
  const defaultProps = {
    messages: [
      {
        id: '1',
        name: 'Test Name',
        email: 'email@domain.com',
        message: 'Test message',
      },
    ],
    columns: tableColumns
  };

  it('renders without crashing', () => {
    shallow(<MessagesTable {...defaultProps} />);
  });

  it('should render default status', () => {
    const tree = renderer
      .create(
        <MessagesTable {...defaultProps} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
