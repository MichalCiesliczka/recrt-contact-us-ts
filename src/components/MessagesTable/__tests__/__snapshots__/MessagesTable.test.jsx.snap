// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessagesTable /> should render default status 1`] = `
<table
  className="MuiTable-root-01"
>
  <thead
    className="MuiTableHead-root-02"
  >
    <tr
      className="MuiTableRow-root-03 MuiTableRow-head-06"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-08 MuiTableCell-head-09"
        scope="col"
      >
        No.
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-08 MuiTableCell-head-09"
        scope="col"
      >
        Name
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-08 MuiTableCell-head-09"
        scope="col"
      >
        Email
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-08 MuiTableCell-head-09"
        scope="col"
      >
        Message
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-016"
  >
    <tr
      className="MuiTableRow-root-03 MuiTableRow-hover-05"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-08 MuiTableCell-body-010"
        scope={undefined}
      >
        1
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-08 MuiTableCell-body-010 Component-column-017"
        scope={undefined}
      >
        Test Name
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-08 MuiTableCell-body-010 Component-column-017"
        scope={undefined}
      >
        email@domain.com
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-08 MuiTableCell-body-010 Component-column-017"
        scope={undefined}
      >
        Test message
      </td>
    </tr>
  </tbody>
</table>
`;
